core/lib/brightcontent/base_controller_ext/pagination.rb

```diff
@@ -5,21 +5,61 @@ module BaseControllerExt
     module Pagination
       extend ActiveSupport::Concern
 
+      included do
+        helper_method :items_per_page
+        helper_method :current_page
+        helper_method :per_page_sizes
+        helper_method :corrected_page
+      end
+
       module ClassMethods
         def per_page_count
-          @per_page_count ||= 30
+          @per_page_count =
+            @per_page_sizes.try(:any?) ? @per_page_sizes[0] : (@per_page_count || 30)
```

Comment:
I think `@page_sizes` can replace the `@per_page_count` variable; an array with one element will have the same effect and can remove the need for nil and empty checks in other places.